src/main/java/org/Cats/SaveManager.java

```diff
@@ -0,0 +1,65 @@
+package org.Cats;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.util.UUID;
+
+public class SaveManager {
+    private File dataDirectory;
+
+    public SaveManager() {
+        this.dataDirectory = new File("data");
+    }
+
+    /**
+     * Creates the data directory if it doesn't already exist
+     */
+    public void createDirectory() {
+        // Create the data directory if it doesn't exist
+        if (!dataDirectory.exists()) {
+            if (!dataDirectory.mkdir()) {
+                // If the directory cannot be created, print an error message
+                try {
+                    System.err.println("Could not create data directory");
+                }catch (RuntimeException e)
+                {
+                    System.out.println("Unable to create data directory");
+                }
+            }
+        }
+    }
+    /**
+     * Saves the cat image to a file
+     * @param catImages an array of CatImage objects to save to a file
+     */
+    public void saveImage(CatImage[] catImages) {
+        try {
+            // Extract the image URL and generate a unique image name
+            String imageUrl = catImages[0].getUrl();
+            String imageName = UUID.randomUUID().toString() + ".jpg";
```

Comment:
This is a nice idea to create unique ID's. However, in this case even if the same image was requested again it would be saved with a different file name.

Reply:
I Changed the the naming method so it will name the image by the extracted id from the request